packages/button/src/Button.ts

```diff
@@ -51,6 +53,45 @@ export class Button extends SizedMixin(ButtonBase) {
         return [...super.styles, buttonStyles];
     }
 
+    protected pendingCooldown = -1;
+
+    constructor() {
+        super();
+        this.addEventListener(
+            'click',
+            (event: Event): void | boolean => {
+                if (this.pending) {
+                    event.preventDefault();
+                    event.stopImmediatePropagation();
+                    event.stopPropagation();
+                    return false;
+                }
+            },
+            {
+                capture: true,
+            }
+        );
+    }
+
+    protected disabledByPending = false;
+
+    @property({ type: Boolean, reflect: true })
+    public set pending(value: boolean) {
+        if (value) {
+            this.pendingCooldown = window.setTimeout(() => {
+                if (!this.disabled) {
+                    this.disabledByPending = true;
+                }
+                this.disabled = true;
+            }, 1000);
+        } else {
+            window.clearTimeout(this.pendingCooldown);
+            if (this.disabledByPending) {
+                this.disabled = false;
+            }
+        }
```

Comment:
Disabling the button does slightly different things to the accessibility tree delivered than I think that we want to here. Name you get "Button, dimmed". We likely want to surface a secondary label as an attribute that we can apply when when `pending`. We're looking at this in #3110 for Picker, so we should make sure the approaches are aligned.

Reply:
Switch to using `is-pending` state, and modified the spectrum-css conversion config to add the right `is-pending` attribute where it finds `disabled`. This is working with a bit of a hack, and I'm not sure of the 'right' fix here, though I think the hack has a low chance of failing?

See here https://github.com/adobe/spectrum-web-components/pull/3163/files#diff-68a09e148bb1fa4125cc905579fa756d38725e86244b153a1ec4c819eaffed5fR44